firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java

```diff
@@ -362,6 +362,18 @@ public WriteBatch batch() {
     return shutdownInternal();
   }
 
+  /**
+   * Wait for server acknowledgement for all pending writes existing at the time of calling this
+   * method.
+   *
+   * <p>Both acceptance and rejection count as server acknowledgement.
+   *
+   * @return A {@link Task} which resolves when all pending writes are acknowledged by the server.
```

Comment:
We likely want to update the comment a bit:

- We should use concepts that are more widely known ("server acknowledgement for all pending writes" -> "when all writes have been sent to the backend").
- We should mention how this behaves during a user change.
- We should mention that this works even after client restart.

If you want to, we can tweak this comment in the chatroom.

Reply:
Yeah. I made some change, but not enough. Especially i don't understand your comment about client restart, and how this still works with a restart. Let's chat tmr.

Reply:
One of the reasons that this is a heavily requested feature is that it offers users the ability to monitor the state of writes that were issued during an earlier app session.If a user issues a write and then closes the app before the write is committed, it is not very straightforward to figure out whether the write made it to the backend once the app is opened again. `drain/awaitPendingWrites` solves this problem nicely.

Just for reference: This can be done today by either issuing a dummy write and waiting for its completion (one would have to know that we commit writes sequentially) or by issuing gets for all documents that were written and checking the `hasPendingWrites` flags.